google/cloud/storage/internal/openssl_util.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/openssl_util.h"
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+
+std::string OpenSslUtils::Base64Decode(std::string const& str) {
```

Comment:
Is there any reason this couldn't be static (and thus go in the header file with the rest of the functions instead)?

Reply:
It is `static`, but that is in the function declaration, in the header file (and can only go there).

I separated the declaration and definition because the function is too big to be inlined (the only motivation to put it in the header file).  I was planning to move the other functions in a future PR too.